integration-testing/tests/avl-producer-api.spec.ts

```diff
@@ -0,0 +1,116 @@
+const avlSubscriptionTableName = `integrated-data-avl-subscription-table-${stage}`;
+
+const testSubscription = {
+    dataProducerEndpoint: "http://siri.ticketer.org.uk/api/vm",
+    description: "Playwright test subscription",
+    shortDescription: "Playwright test subscription",
+    username: "PLAYWRIGHT",
+    password: "PLAYWRIGHT",
+    subscriptionId: "PLAYWRIGHT",
+    requestorRef: "BODS_MOCK_PRODUCER",
+    publisherId: "PLAYWRIGHT",
+};
+
+const cleardownTestSubscription = async () => {
+    await deleteDynamoItem(avlSubscriptionTableName, {
+        PK: testSubscription.subscriptionId,
+        SK: "SUBSCRIPTION",
+    });
+};
+
+test.beforeAll(async () => {
+    await cleardownTestSubscription();
+});
+
+test.afterAll(async () => {
+    await cleardownTestSubscription();
+});
+
+test.describe("avl-producer-api", () => {
+    test("should allow a new data producer subscription to be created", async ({ request }) => {
+        const newSubscription = await request.post(`${avlProducerApiUrl}/subscriptions`, {
+            data: testSubscription,
+        });
+
+        expect(newSubscription.status()).toBe(201);
+
+        const getSubscriptionResponse = await request.get(
+            `${avlProducerApiUrl}/subscriptions/${testSubscription.subscriptionId}`,
+        );
+
+        expect(getSubscriptionResponse.status()).toBe(200);
+
+        expect(await getSubscriptionResponse.json()).toEqual(
+            expect.objectContaining({
+                id: testSubscription.subscriptionId,
+                publisherId: testSubscription.publisherId,
+                status: "LIVE",
+            }),
+        );
+
+        const listSubscriptionsResponse = await request.get(`${avlProducerApiUrl}/subscriptions/`);
+
+        expect(listSubscriptionsResponse.status()).toBe(200);
+    });
+
+    test("should allow an existing data producer subscription to be updated", async ({ request }) => {
+        const updateSubscription = await request.put(
+            `${avlProducerApiUrl}/subscriptions/${testSubscription.subscriptionId}`,
+            {
+                data: { ...testSubscription, username: "newUsername", password: "newPassword" },
+            },
+        );
+
+        expect(updateSubscription.status()).toBe(204);
+
+        const getSubscriptionResponse = await request.get(
+            `${avlProducerApiUrl}/subscriptions/${testSubscription.subscriptionId}`,
+        );
+
+        expect(getSubscriptionResponse.status()).toBe(200);
+
+        expect(await getSubscriptionResponse.json()).toEqual(
+            expect.objectContaining({
+                id: testSubscription.subscriptionId,
+                publisherId: testSubscription.publisherId,
+                status: "LIVE",
+            }),
+        );
+
+        const listSubscriptionsResponse = await request.get(`${avlProducerApiUrl}/subscriptions/`);
+
+        expect(listSubscriptionsResponse.status()).toBe(200);
+    });
```

Comment:
Same as previous comment - would be good to assert that test subscription is in the subscription list response